rules.py

```diff
@@ -826,7 +826,53 @@ def check(self):
         if error_log:
             raise base.ElectionTreeError(
                 "The Election File contains unreferenced Candidates", error_log)
-                
+
+class CheckIdentifiers(base.TreeRule):
+    """Check that the NIST objects in the feed has an '<ExternalIdentifier>' block.
+	Add error message if the block is missing."""
+    
+    identifier_values = {}
+    error_log = []
+    nist_objects = ('Candidate', 'Contest', 'Party')
+    def check(self):
+        for event, element in etree.iterwalk(self.election_tree):
+	    tag = self.strip_schema_ns(element)
+            if tag not in self.nist_objects:
+                continue
+            else:
+                nist_obj = tag
+                object_id = element.get('objectId')
+                external_identifiers = element.find("ExternalIdentifiers")
+                if external_identifiers is None:
+                    error_message = ("Missing <ExternalIdentifiers>"
+			" tag/block in {0} with ID {1}.".format(nist_obj, object_id))
+                    self.error_log.append(base.ErrorLogEntry(
+                            element.sourceline, error_message))
+                else:
+                    identifier = external_identifiers.find("ExternalIdentifier")
+                    if identifier is None:
```

Comment:
You don't need to do this check since `ExternalIdentifer` must appear at least once in an `ExternalIdentifiers` block. Otherwise it's a schema error and will be caught by the schema validation stage.